sync/options.go

```diff
@@ -21,18 +21,19 @@
 package sync
 
 import (
+	"runtime"
 	"time"
 
 	"github.com/m3db/m3x/instrument"
 )
 
 const (
-	defaultNumShards             = int64(2 ^ 4)
-	defaultKillWorkerProbability = 0.0001
+	defaultKillWorkerProbability = 0.001
 	defaultGrowOnDemand          = false
 )
 
 var (
+	defaultNumShards int64 // Set by init().
```

Comment:
Why not just initialize this to `int64(runtime.NumCPU())` here?

Reply:
for some reason I thought I couldn't call a function, will fix